### PR TITLE
Add ReadNetFromONNX and ReadNetFromONNXBytes

### PR DESCRIPTION
**Pull request by JoshChristie**
Simply adds ReadNetFromONNX and ReadNetFromONNXBytes

### CODE REVIEW

File: dnn_test.go [hunk 1]
```diff
@@ -283,6 +283,81 @@ func TestTensorflowMemory(t *testing.T) {
 	checkTensorflowNet(t, net)
 }
 
+func TestOnnxMemory(t *testing.T) {
+	path := os.Getenv("GOCV_ONNX_TEST_FILES")
+	if path == "" {
+		t.Skip("Unable to locate ONNX model file for tests")
+	}
+
+	b, err := ioutil.ReadFile(path + "/googlenet-9.onnx")
```
Comment (OmriTreidel) [inline dnn_test.go]: it might be cleaner to use `filepath.Join(path, "googlenet-9.onnx")`

File: dnn_test.go [hunk 2]
```diff
@@ -283,6 +283,81 @@ func TestTensorflowMemory(t *testing.T) {
...
+	}
+
+	net, err := ReadNetFromONNXBytes(b)
+	if err != nil {
+		t.Errorf("Failed to load Tensorflow model from bytes: %v", err)
+	}
+	if net.Empty() {
+		t.Errorf("Unable to load Tensorflow model")
+	}
+	defer net.Close()
```
Comment (OmriTreidel) [inline dnn_test.go:305]: may as well remove the defer and simply close the net after checkONNXNet

File: dnn_test.go [hunk 3]
```diff
@@ -283,6 +283,81 @@ func TestTensorflowMemory(t *testing.T) {
...
+	}
+	defer img.Close()
+
+	blob := BlobFromImage(img, 1.0, image.Pt(224, 224), NewScalar(0, 0, 0, 0), true, false)
+	if blob.Empty() {
+		t.Error("Invalid blob in ONNX test")
+	}
+	defer blob.Close()
+
+	net.SetInput(blob, "data_0")
```
Comment (OmriTreidel) [inline dnn_test.go:338]: are "data_0" and :prob_1" names of the specific network you're testing?

Comment (JoshChristie) [reply dnn_test.go:338]: Yes, it's the input and output names specific to this model

File: dnn_test.go [hunk 4]
```diff
@@ -283,6 +283,81 @@ func TestTensorflowMemory(t *testing.T) {
...
+	checkONNXNet(t, net)
+}
+
+func TestOnnxDisk(t *testing.T) {
+	path := os.Getenv("GOCV_ONNX_TEST_FILES")
+	if path == "" {
+		t.Skip("Unable to locate ONNX model file for tests")
+	}
+
+	net := ReadNetFromONNX(path + "/googlenet-9.onnx")
```
Comment (OmriTreidel) [inline dnn_test.go]: filepath

File: dnn_test.go (hunk 2 above)
Comment (JoshChristie) [reply dnn_test.go:305]: it's safe this way IMO

**Review by OmriTreidel — CHANGES_REQUESTED** [review]
Looks good. Just a few minor suggestions/comments

**Comment by deadprogram** [comment]
Looks very good, thank you very much @JoshChristie and also to @OmriTreidel for reviewing.

Now squash/merging.

**Pull request merged**